OracleGoldenGate/21c/bin/deployment-main.sh

```diff
@@ -125,34 +130,69 @@ function setup_deployment_directories() {
-## If defined, the command identified by ${RUN_COMMAND} will be run.
-## 
-## The value of ${RUN_COMMAND_BACKGROUND} by default is set to true.
-##     When ${RUN_COMMAND_BACKGROUND} is true:
-##          The command identified by ${RUN_COMMAND} will be run in the background 
-##          and its return status is ignored
-##     When ${RUN_COMMAND_BACKGROUND} is false:
-##          The command identified by ${RUN_COMMAND} will be run in the foreground 
-##          and must return a status of 0 in order to proceed with the start sequence
-##
-function run_prestart_command {
-    [[ ! -z "${RUN_COMMAND}" ]] && {
-
-        case "${RUN_COMMAND_BACKGROUND,,}" in
-            true)
-                ${RUN_COMMAND} &
-                ;;
-
-            false)
-                ${RUN_COMMAND} || abort "${RUN_COMMAND} did not complete successfully!"
-                ;;
-            *)
-                abort "Unknown RUN_COMMAND_BACKGROUND state [${RUN_COMMAND_BACKGROUND}] for run_prestart_command: expected true/false"
-        esac
+##  r u n _ u s e r _ s c r i p t s
+##
+## Hook for launching custom scripts in the container before and after ogg start
+##     Default Values:
+##       - ${ABORT_ON_USER_SCRIPT_ERRORS} : true
+##       - ${SETUP_USER_SCRIPTS}          : "${OGG_HOME}/scripts/setup"
+##       - ${STARTUP_USER_SCRIPTS}        : "${OGG_HOME/scripts/startup}"
+##
+## Scripts are run lexicographically and recursively from the directories pointed to by:
+##      - ${SETUP_USER_SCRIPTS} are executed prior to any other steps in the boot sequence
+##      - ${STARTUP_USER_SCRIPTS} are executed after ogg/nginx startup
+##
+## When ${ABORT_ON_USER_SCRIPT_ERRORS} is true:
+##    Scripts must return a status of 0 on exit or the boot proccess will abort
+##
+## When ${ABORT_ON_USER_SCRIPT_ERRORS} is false:
+##    Scripts that return a non-zero status will issue a warning but will continue
+##    the boot process
+##
+function run_user_scripts {
+
+    local SCRIPTS_ROOT="${1}";
+    
+    [ -z "$SCRIPTS_ROOT" ] && {
+        printf "%s: No SCRIPTS_ROOT passed on, no scripts will be run\n" "${0}";
+        return 1;
+    } || {
+        if [ -d "$SCRIPTS_ROOT" ] && [ -n "$(ls -A "$SCRIPTS_ROOT")" ]; then
+            printf "Executing user defined scripts in: %s\n" "${SCRIPTS_ROOT}"
+        
+            for f in "${SCRIPTS_ROOT}"/*; do
+                [ -d "${f}" ] && {
+                    run_user_scripts "${f}"
+                } || {
+                    case "$f" in
+                        *.sh)
+                                printf "%s: running %s\n" "${0}" "${f}";
+                                . "$f" && state=$? || state=$?
+                                ;;
+                        *.sql)
+                                printf "%s: running %s\n" "${0}" "${f}";
+                                echo "exit" | LD_LIBRARY_PATH=$(dirname ${SQLPLUS}):${LD_LIBRARY_PATH} ${SQLPLUS} -s "/ as sysdba" @"$f" && state=$? || state=$?
```

Comment:
We can't execute SQL*Plus from here because the database is likely not in the same container. When running in another container, credentials are needed, etc.

Because of the inherent complexities, I would like to see support for `*.sql' files removed.